src/ast/query.rs

```diff
@@ -64,6 +64,23 @@ impl fmt::Display for Query {
     }
 }
 
+/// Options for SetOperator
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub enum SetOperatorOption {
+    All,
+    Distinct,
+}
+
+impl fmt::Display for SetOperatorOption {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            SetOperatorOption::All => write!(f, " ALL"),
+            SetOperatorOption::Distinct => write!(f, " DISTINCT"),
```

Comment:
Printing spaces inside the enum structure doesn't make much sense.

If there's a space or not, the structure that holds it should handle it (in this case, the SetExpr).

Reply:
I fixed this at [f6e8305](https://github.com/sqlparser-rs/sqlparser-rs/pull/689/commits/f6e83058b0b5a37de5d2c9b0ce473186a1294ed4)